src/content/reference/manifest.mdx

```diff
@@ -509,6 +509,31 @@ metadata:
     custom.label/dev: "true"
 ```
 
+## mode (string, optional)
+
+Defines how `okteto up` will setup the development container and behave within the cluster.
+
+Currently two options are available:
+
+- `sync`: (default) Sync & run your code in a remote development container.
+- `hybrid`: Run your code on your local computer instead of in a remote development container, while rest of your service components runs in your remote cluster.
+
+|                   | `sync` (default) | `hybrid` |
+| :---------------- | :--------------- | :------- |
+| Source code       | Source code is edited locally and synchronizes bidirectionally (via Syncthing) with a copy of the source code in the remote development container. | Source code is edited locally for a chosen set of microservices. |
+| App location      | The application exists entirely within the remote development container. | Your application is split between your local environment and the remote development container based on your configuration. |
+| `command` context | CLI commands run in the remote development container. | CLI commands run on your local machine, so traffic need to be rerouted to/from the cluster, e.g. using the reverse option |
+
+```yaml
+dev:
+  frontend:
+    mode: hybrid
+    reverse:
+    - 3000:8080
+    workdir: ./frontend
+    command: ["npx webpack watch --mode development"]
+```
+
```

Comment:
With the new `hybrid` not all parameters in the `dev` section apply in this mode. Shouldn't we add a clarification in every one of these parameters stating if it's supported in `sync` and/or `hybrid` mode?

Example:

```
### envFiles
> Supported in both `sync` and `hybrid` modes.
...

### affinity
> Only supported in `sync` mode.
...
```

Reply:
@rlamana Ah yes. I think that's a good and important clarification to add.

Are those the only two exceptions or do we need to compile a complete list?

Reply:
Not, we have to take into account all of them. The list is compiled in the spec doc here: https://www.notion.so/okteto/Okteto-CLI-3-0-7fbd3a6a566145d3a4930d3c15fc642d?pvs=4#37aeeb1c05824203ac5f8132caf76e1f

@andreafalzetti @teresaromero you might have more context if that table there is updated after the work done there or is there anything we need to change?

Reply:
@omnomagonz @rlamana 

These are the fields that are supported in `sync` but not in `hybrid`:

- affinity
- context
- externalVolumes
- image
- imagePullPolicy
- initContainer
- initFromImage
- lifecycle
- namespace
- nodeSelector
- persistentVolume
- push
- replicas
- secrets
- securityContext
- serviceAccount
- sshServerPort
- sync
- tolerations
- volumes


These fields work in both modes:

- annotations
- autocreate
- command
- container
- envFiles
- environment
- environment
- forward
- metadata
- mode
- reverse
- selector
- workdir

These fields work in `sync` but we're not sure if they make sense for `hybrid`:
- interface
- probes
- remote
- timeout

This field exists in code but not documented and not sure if it should be supported:
- args

Should work in both but deprecated:
- healthchecks
- labels

Reply:
Thanks a lot @andreafalzetti for the updated list. We will need to clarify those that we are not sure:
- interface
- probes
- remote
- timeout
As discussed offline maybe @pchico83 can help us with more context on these, and soon before he leaves on PTO.

Reply:
Thank you both for this context 🙇‍♂️ I'll commit a change with how I propose we represent this and let's continue iterating 🙂

Reply:
For clarity:

> These fields work in sync but we're not sure if they make sense for hybrid:

Does this mean those options work in both but only are compatible/make sense for `sync`? So a user could use these options in `hybrid` mode but we're not sure how they'll work or how they'll benefit the user?

Reply:
Resolving this conversation; please see latest commit: https://github.com/okteto/docs/pull/450/commits/d45fc31ba9222085aca74aa816e2e4cdbbe00831 and add additional comments :)

I left out some of the nuanced details like "deprecated but works in both" and "exists in code but not documented". We can iterate to add more clarity on these but I didn't want to get blocked on nuance for now

Reply:
@omnomagonz "not sure if they make sense for hybrid" meant that we are still deciding if the will fall under the "ignored/unsupported" fields or if we will implement it in both modes. But there will be a final resolution for all those so that the final list will only have "supported or not supported" for each mode.

@andreafalzetti what is the current status of those fields?

Reply:
In my opinion, besides adding a table I will still add a "hint" on every single parameter. It is common to just go and check a parameter in the docs to learn about it. If the user doesn't have that info there (and assuming they are aware of these two modes and know there is a tablefor it), users will have to go to look for the table and see if the parameter is there, and then maybe go back to the parameter description to continue reading about it. I find the table less important since, you usually go seeking for documentation on a particular item. It is pretty common to see this type of support descriptions in docs of this kind.

Reply:
@rlamana for `remote` I was able to verify that it works for hybrid too. `timeout` for me doesn't work in sync either, so I've created https://github.com/okteto/okteto/issues/3849

still not sure about these tbh:
- interface
- probes

as soon as I can verify I let you guys know.

Reply:
@andreafalzetti So after our daily today, I think `probes` was clarified and it's good to go. What about `interface` then?